popde/density_estimate.py

```diff
@@ -421,6 +424,24 @@ def evaluate_KDEpy(self, points):
 
         return density_values
 
+    def symmetrize_data(self, dims):
+        """
+        Override parent method to also handle per-point bandwidth arrays.
+        Augment kde_data with a copy exchanging values between the two indexed
+        dimensions, and also double the bandwidth array if it's per-point.
+        """
+        # Call parent's symmetrize_data to handle data and weights
+        super().symmetrize_data(dims)
+
+        # Now handle bandwidth if it's an array
+        # Check if bandwidth is array-like (not a scalar)
+        if isinstance(self.bandwidth, (list, tuple, np.ndarray)):
+            bw_array = np.asarray(self.bandwidth)
+            # Only double if it's a 1D array matching original data length
```

Comment:
If it's not 1-dimensional then I don't know what the code can do with it. It should always be 1d, so can we just do eg `assert bw_array.ndim == 1` immediately after defining bw_array ?